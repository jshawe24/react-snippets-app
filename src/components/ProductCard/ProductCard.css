/* src/components/ProductCard/ProductCard.css */

.product-card {
    border: 1px solid #ccc;
    border-radius: 8px;
    box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);
    padding: 20px;
    margin: 20px;
    transition: transform 0.3s, box-shadow 0.3s;
    background-color: #ffffff;
    max-width: 300px;
}

.product-card:hover {
    transform: translateY(-5px);
    box-shadow: 0 8px 16px rgba(0, 0, 0, 0.2);
}

.product-card h2 {
    font-size: 1.5em;
    color: #333;
}

.product-card h3 {
    font-size: 1.2em;
    color: #007BFF;
}

.product-card p {
    color: #666;
    margin: 10px 0;
}

.product-card .price {
    font-weight: bold;
    color: #28a745;
}

.limited-stock {
    color: #dc3545 !important; /* Red for limited stock */
    font-weight: bold;
}

.buy-now {
    color: #007BFF !important; /* Blue for call to action */
    font-weight: bold;
}

@media (max-width: 600px) {
    .product-card {
        max-width: 90%;
        margin: 10px auto;
    }
}
