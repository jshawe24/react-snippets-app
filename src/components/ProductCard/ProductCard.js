// src/components/ProductCard/ProductCard.js
import React from 'react';
import './ProductCard.css'; // Import the CSS file

const ProductCard = ({quantity}) => {
  return (
    <div>
            <h2>Product Card</h2>
            <h3>Scenario</h3>
            <p>
                Set text and the style of the text depending on a quantity value passed through to the component.
            </p>

            <h3>Info</h3>
            <p>
                In this scenario, we have a quantity parameter passed through from our App.js routing. We then evaluate the value and set both style and text depending on the value, using a tenary operator
            </p>

            <h3>Testing</h3>
            <div>
                <p>
                    To test this, change the quantity value in App.js - <br></br>
                    <i style={{ color: 'blue' }}>
                    {'<Route path="/product-card" element={<ProductCard quantity="2" />} />'}
                    </i>
                </p>
            </div>


            <div className="product-card">
                <h3>Product Name</h3>
                <p>Description of the product.</p>
                <p className="price">Price: $99.99</p>
                <p>Quantity {quantity}</p>
                <p className={quantity < 5 ? "limited-stock" : "buy-now"}>
                    {quantity < 5 ? "Limited stock" : "Available buy now"}
                </p>
            </div>
    </div>
  );
};

export default ProductCard;
