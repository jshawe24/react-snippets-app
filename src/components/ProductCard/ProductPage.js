// src/components/ProductCard/ProductPage.js

import React from 'react';
import ProductCard from './ProductCard';

const ProductPage = () => {
    return (
        <div>
            <h2>Product Card</h2>

            <h3>Scenario</h3>
            <p>
                Setting text based on parameter "quantity" passed through to the component.
            </p>

            <h3>Info</h3>
            <p>
                In this scenario, we have a quantity parameter passed through to our ProductCard component. We then evaluate the quantity and set both style and text depending on the value, using a ternary operator.
            </p>

            <h3>Testing</h3>
            <p>
                To test this, change the quantity value in App.js - <br />
                <i style={{color: 'blue'}}>
                    {'<ProductCard quantity={2} /><ProductCard quantity={10} />'}
                </i>
            </p>

            {/* Render multiple ProductCards with different quantities */}
            <ProductCard productName="Jumpers" quantity={2} />
            <ProductCard productName="Trousers" quantity={10} />
        </div>
    );
};

export default ProductPage;
