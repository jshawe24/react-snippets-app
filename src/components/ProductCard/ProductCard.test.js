import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom'; // Correct import without '/extend-expect'
import ProductCard from './ProductCard';

describe('ProductCard Component', () => {

  test('renders product name', () => {
    render(<ProductCard productName="Test Product" quantity={10} />);
    const productNameElement = screen.getByText(/Test Product/i);
    expect(productNameElement).toBeInTheDocument();
  });

  test('renders quantity', () => {
    render(<ProductCard productName="Test Product" quantity={2} />);
    const quantityElement = screen.getByText(/Quantity 2/i);
    expect(quantityElement).toBeInTheDocument();
  });

  test('renders limited stock message when quantity is less than 5', () => {
    render(<ProductCard productName="Test Product" quantity={3} />);
    const limitedStockElement = screen.getByText(/Limited stock/i);
    expect(limitedStockElement).toBeInTheDocument();
    expect(limitedStockElement).toHaveClass('limited-stock');
  });

  test('renders available message when quantity is 5 or more', () => {
    render(<ProductCard productName="Test Product" quantity={5} />);
    const buyNowElement = screen.getByText(/Available buy now/i);
    expect(buyNowElement).toBeInTheDocument();
    expect(buyNowElement).toHaveClass('buy-now');
  });
  
  test('matches snapshot', () => {
    const { asFragment } = render(<ProductCard productName="Test Product" quantity={10} />);
    expect(asFragment()).toMatchSnapshot();
  });

});