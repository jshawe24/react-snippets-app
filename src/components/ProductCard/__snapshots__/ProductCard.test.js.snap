// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard Component matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="product-card"
    >
      <h3>
        Test Product
      </h3>
      <p>
        Description of the product.
      </p>
      <p
        class="price"
      >
        Price: $99.99
      </p>
      <p>
        Quantity 10
      </p>
      <p
        class="buy-now"
      >
        Available buy now
      </p>
    </div>
  </div>
</DocumentFragment>
`;
